File: rb/features.rb
# YoMama SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module YoMamaFeatures
  def self.make_feature(name)
    case name
    when "base"
      YoMamaBaseFeature.new
    when "test"
      YoMamaTestFeature.new
    else
      YoMamaBaseFeature.new
    end
  end
end

File: py/features.py
# YoMama SDK feature factory

from feature.base_feature import YoMamaBaseFeature
from feature.test_feature import YoMamaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YoMamaBaseFeature(),
        "test": lambda: YoMamaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
